sycl/test-e2e/Basic/built-ins/marray_math.cpp

```diff
@@ -0,0 +1,139 @@
+        std::uint64_t result_uint64;                                           \
+        std::memcpy(&result_uint64, &result[i], sizeof(result[i]));            \
+        std::ostringstream stream;                                             \
+        stream << "0x" << std::hex << result_uint64;                           \
+        std::string result_string = stream.str();                              \
+        result_string = result_string.substr(result_string.size() - 5);        \
+        assert(result_string.compare(EXPECTED[i]) == 0);                       \
+      }                                                                        \
+    }                                                                          \
+  }
+
+#define EXPECTED(TYPE, ...) ((TYPE[]){__VA_ARGS__})
+
+int main() {
+  sycl::queue deviceQueue;
+
+  sycl::marray<float, 2> ma1{1.0f, 2.0f};
+  sycl::marray<float, 2> ma2{3.0f, 2.0f};
+  sycl::marray<float, 3> ma3{180, 180, 180};
+  sycl::marray<int, 3> ma4{1, 1, 1};
+  sycl::marray<float, 3> ma5{180, -180, -180};
+  sycl::marray<float, 3> ma6{1.4f, 4.2f, 5.3f};
+  sycl::marray<unsigned int, 3> ma7{1, 2, 3};
+  sycl::marray<unsigned long int, 3> ma8{1, 2, 3};
+
+  TEST(sycl::fabs, float, 3, EXPECTED(float, 180, 180, 180), 0, ma5);
+  TEST(sycl::ilogb, int, 3, EXPECTED(int, 7, 7, 7), 0, ma3);
+  TEST(sycl::fmax, float, 2, EXPECTED(float, 3.0f, 2.0f), 0, ma1, ma2);
+  TEST(sycl::fmax, float, 2, EXPECTED(float, 5.0f, 5.0f), 0, ma1, 5.0f);
+  TEST(sycl::fmin, float, 2, EXPECTED(float, 1.0f, 2.0f), 0, ma1, ma2);
+  TEST(sycl::fmin, float, 2, EXPECTED(float, 1.0f, 2.0f), 0, ma1, 5.0f);
+  TEST(sycl::ldexp, float, 3, EXPECTED(float, 360, 360, 360), 0, ma3, ma4);
+  TEST(sycl::ldexp, float, 3, EXPECTED(float, 5760, 5760, 5760), 0, ma3, 5);
+  TEST(sycl::pown, float, 3, EXPECTED(float, 180, 180, 180), 0.1, ma3, ma4);
+  TEST(sycl::pown, float, 3, EXPECTED(float, 180, 180, 180), 0.1, ma3, 1);
+  TEST(sycl::rootn, float, 3, EXPECTED(float, 180, 180, 180), 0.1, ma3, ma4);
+  TEST(sycl::rootn, float, 3, EXPECTED(float, 2.82523, 2.82523, 2.82523),
+       0.00001, ma3, 5);
+  TEST2(sycl::fract, float, float, 3, EXPECTED(float, 0.4f, 0.2f, 0.3f),
+        EXPECTED(float, 1, 4, 5), 0.0001, ma6);
+  TEST2(sycl::modf, float, float, 3, EXPECTED(float, 0.4f, 0.2f, 0.3f),
+        EXPECTED(float, 1, 4, 5), 0.0001, ma6);
+  TEST2(sycl::sincos, float, float, 3,
+        EXPECTED(float, 0.98545f, -0.871576f, -0.832267f),
+        EXPECTED(float, 0.169967, -0.490261, 0.554375), 0.0001, ma6);
+  TEST2(sycl::frexp, float, int, 3, EXPECTED(float, 0.7f, 0.525f, 0.6625f),
+        EXPECTED(int, 1, 3, 3), 0.0001, ma6);
+  TEST2(sycl::lgamma_r, float, int, 3,
+        EXPECTED(float, -0.119613f, 2.04856f, 3.63964f), EXPECTED(int, 1, 1, 1),
+        0.0001, ma6);
+  TEST2(sycl::remquo, float, int, 3, EXPECTED(float, 1.4f, 4.2f, 5.3f),
+        EXPECTED(int, 0, 0, 0), 0.0001, ma6, ma3);
+
+  if (!deviceQueue.get_device().is_gpu()) {
```

Comment:
Note for reviewers: nan provides wrong result on GPU, disabling it for now, internal ticket is going to be created.